src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs

```diff
@@ -356,6 +356,18 @@ internal void ParseStream(BufferedReadStream stream, SpectralConverter spectralC
                 // to uint to avoid sign extension.
                 if (stream.RemainingBytes < (uint)markerContentByteSize)
                 {
+                    if (metadataOnly && this.Metadata != null && this.Frame != null)
+                    {
+                        // We have enough data to decode the image, so we can stop parsing.
+                        return;
+                    }
+
+                    if (this.Metadata != null && this.Frame != null && spectralConverter.HasPixelBuffer())
```

Comment:
As we want to now if image was decoded at this point - we can use `Converted` of `SpectralConverter` as it's called right after **baseline** image is decoded. 

Two problems:

1. It's `protected`, not really a problem creating a public getter though
2. It's not set for progressive images as those are lazy-converted during `pixel buffer -> Image<T>` step which won't be called until parsing loop is finished, not the wisest solution from me

I'd probably refactor spectral converter so it could better describe its state.

Reply:
Yeah. This felt a little hacky. State matching would be better.